src/sockaddr.rs

```diff
@@ -153,6 +143,26 @@ impl SockAddr {
         })
     }
 
+    /// Constructs a `SockAddr` with the family `AF_UNIX` and the provided path.
+    ///
+    /// Returns an error if the path is longer than `SUN_LEN`.
+    pub fn unix<P>(path: P) -> io::Result<SockAddr>
+    where
+        P: AsRef<Path>,
+    {
+        crate::sys::unix_sockaddr(path.as_ref())
+    }
+
+    /// Set the length of the address.
+    ///
+    /// # Safety
+    ///
+    /// Caller must ensure that the address up to `length` bytes are properly
+    /// initialised.
+    pub unsafe fn set_length(&mut self, length: socklen_t) {
+        self.len = length;
+    }
```

Comment:
This puts a libc type in your public API. Did you intend to do that?

Reply:
We already have that in various other places, e.g. the `len` function also returns `socklen_t`. So, it's not ideal and I'm aware of the complexities that then come with updating libc and windows-sys, but they're type definitions (not e.g. structs) should they should remain compatible between dependency updates.